### ISSUE TITLE
Document dev mode not running eager bindings on start

### ISSUE THREAD

**Issue by hoxu** [playframework/playframework#10469]
[Scala Dependency Injection / Eager bindings](https://www.playframework.com/documentation/2.8.x/ScalaDependencyInjection#Eager-bindings):

> If you want to create them once and also create them eagerly **when the application starts up**, rather than lazily when they are needed, then you can Guice’s eager singleton binding.
> ...
> Eager singletons can be used to start up a service **when an application starts**.

However, there is no mention that dev mode does not create eager bindings until the first request. It's mentioned in a 2016 thread here:

https://groups.google.com/g/play-framework/c/21G3eEU-3Y8

I think the eager bindings chapter would be a good spot to mention this.

### Play Version

2.8.2

### API

Scala

**Comment by wsargent** [comment]
I don't think this is right.  It is new eager bindings on start, but the new application doesn't start until there's a new request -- it's request driven.

**Comment by RommelTJ** [comment]
@hoxu @wsargent I can confirm what Will said is correct. Eager bindings are request driven. 

I'm adding documentation here to hopefully clarify this: https://github.com/playframework/playframework/pull/12896
